install.sh

```diff
@@ -829,11 +829,27 @@ do_unmount_and_remove() {
     set -x
 }
 
+clean_mounted_directory() {
+    for path in "$1"/*; do
+        if [ -d "$path" ]; then
+            if grep -q "$path" /proc/mounts; then
+                clean_mounted_directory "$path"
+            else
+                rm -rf "$path"
+            fi
+        else
+            rm "$path"
+        fi
+     done
+}
+
 do_unmount_and_remove '/run/k3s'
-do_unmount_and_remove '/var/lib/rancher/k3s'
+do_unmount_and_remove '/var/lib/rancher/k3s/data'
+do_unmount_and_remove '/var/lib/rancher/k3s/storage'
 do_unmount_and_remove '/var/lib/kubelet/pods'
 do_unmount_and_remove '/var/lib/kubelet/plugins'
 do_unmount_and_remove '/run/netns/cni-'
+clean_mounted_directory '/var/lib/rancher/k3s'
```

Comment:
wait wait wait, we put this in the killall script... so now whenever you run the killall script it deletes all of K3s off your node. This makes sense to do in the uninstaller but the killall script DEFINITELY shouldn't be deleting everything out from under /var/lib/rancher/k3s otherwise you end up wiping your cluster off the face of the earth when all you wanted to do was kill the pods and service.

Reply:
@brandond when you said about deleting everything inside the `/var/lib/rancher/k3s` means also for the `data` and `storage`? btw I added the `clean_mounted_directory` to the uninstall script.

Reply:
thanks! Sorry, I realize my review here has not been great, I forgot that we were talking about killall and not just uninstall.